GlobalGameJam2018Alchemy/Assets/Scripts/GameController.cs

```diff
@@ -25,8 +28,10 @@ public IEnumerator SpawnIngredients()
         {
             if (!GetComponent<NetworkController>().IsSinglePlayer)
             {
-                // we always need to return something, otherwise an endless loop might occur.
-                yield return new WaitForSeconds(singlePlayerIgredientSpawnInterval);
+                // Ingredients are coming via network and thus we only keep this coroutine running
+                // if later a singleplayer game might be started.
+                yield return new WaitForSeconds(multiPlayerIngredientSpawnSleepInterval);
```

Comment:
Not sure if that is a good idea, because a player might have to wait 1min for ingredients when switching from multiplayer to singleplayer.
Maybe we'll have to rework the code to start the routine in singleplayer and stop it everywhere else (menu and multiplayer).

Reply:
It would definitely be the cleanest solution to not even get into this coroutine in multiplayer and only have it in singleplayer. And since we are no longer at the game jam, perhaps we should really try to go for the best solution :)

I'll have a look at it and see if I can come up with a solution.